### DISCUSSION TITLE
Shouldn't main's generated args be const?

### DISCUSSION THREAD

**Discussion by KendallHarter** [hsutter/cppfront#764] [Q&A]
Currently, the `args` generated by cppfront in main isn't const ([see here](https://github.com/hsutter/cppfront/blob/main/source/cppfront.cpp#L4343)) which seems to create an inconsistency within the language.
For example, the following will compile because `args` isn't const:
```
main: (args) = {
    args.pop_back();
}
```

The seemingly identical following code will fail to compile, though, because `args` will be const here
```
not_main: (args: std::vector<std::string_view>) = {
    args.pop_back();
}
```

Is this an intentional design decision/special case for main?  As far as my understanding goes, it shouldn't be mutable as it's an `in` parameter, but maybe I'm missing something.

**Comment by realgdman** [comment]
(At least in C, argv is mutable and can be used for scratchpad memory.)
Nvm, found out there is args.mutable_argv for that.

**Comment by JohelEGP** [comment]
I'd say it's an accident.
There's plans to make it a model of `std::ranges::view` to avoid allocations.

**Comment by hsutter** [comment]
Thanks! Good point, this was an oversight.

**Comment by hsutter** [comment]
Fixed: 32a72fcdf1faa4cb5d2ca81584a40078e4183b21
